### ISSUE TITLE
response not being generated 

### ISSUE THREAD

**Issue by chetanr2512** [zinedkaloc/ai-travel-planner#1]
So I followed all the steps as it was in the documentation but when I select all the options on the screen and click on generate it shows the hourglass for a while and then nothing is show. Would like to know where I am wrong

**Comment by zinedkaloc** [comment]
Thanks for sharing the issue you are experiencing.

One suggestion to try would be to use a different browser. You can try using a different browser and see if the issue persists.

Additionally, it's also possible that the generation process may take some time to complete, depending on the amount of data you are generating. However, since you mentioned that nothing is shown after the hourglass, it's likely that there could be an issue with the response.

Another suggestion is to check for any error messages in the browser console. To do this, you can open the browser console and see if there are any error messages that can help you identify the issue. This can provide more information in diagnosing the issue and finding a solution. Let me know if you have any questions or need further assistance.

**Comment by chetanr2512** [comment]
sure man thanks for the quick reply

**Comment by chetanr2512** [comment]
these are the errors i got 
![Screenshot 2023-04-14 225124](https://user-images.githubusercontent.com/130687370/232115097-6c6194fb-5932-4ca3-8778-a1d55a321800.png)

**Comment by zinedkaloc** [comment]
Thanks for letting me know about the error message. 

It looks like the response is returning a 400 error. To troubleshoot this issue, I suggest creating your own service and copying the endpoint URL from Altogic. Once you have the URL you can add it to your app's .env file as REACT_APP_ENDPOINT_URL. This should help resolve the issue.

**Comment by chetanr2512** [comment]
i did the same thing I created my own service I mean to say I followed your instructions from here https://www.altogic.com/blog/travel-planner-with-openai-chatgpt-api#building-the-backend
and made the service .
Do you want me to create a different service cause I don't know how

**Comment by zinedkaloc** [comment]
Could you please check https://watcher.altogic.com/ to see if there are any error messages or additional information that might be helpful? 

It appears that you are receiving a 400 error response, but taking a closer look at the logs or any other details on the site could provide more insights into the issue. Let me know if you find anything, and I'll do my best to help you resolve the issue.

**Comment by chetanr2512** [comment]
this was in the watcher under the endpoint tab
![Screenshot 2023-04-14 232218](https://user-images.githubusercontent.com/130687370/232120507-d166d55e-c76e-4d00-9fb8-38e421000c60.png)
and this is the action detail like the first few lines

{
  "timestamp": "2023-04-14T17:47:07.476Z",
  "level": "error",
  "type": "api",
  "action": "process-endpoint",
  "cluster": "c5-na",
  "hostName": "engine-deployment-86d5dc897d-v62kx",
  "processId": 18,
  "envId": "64397ef3e509294a84918416",
  "keyId": null,
  "userId": "64397e35aab102e1206edfca",
  "appId": "64397ef3e509294a84918407",
  "endpointId": "64398062aab102e1206ee7ea",
  "endpointInfo": {
    "endpointId": "64398062aab102e1206ee7ea",
    "name": "Travel itinerary endpoint",
    "iname": "64398062aab102e1206ee7e9",
    "path": "/travel",
    "method": "POST"
  }

**Comment by zinedkaloc** [comment]
To better understand the issue, you may want to take a look at the rest of the JSON response and see if there are any error messages or specific details that can help identify the cause of the problem. 

Specifically, you can check the `response.errors` section of the JSON to get more information.

**Comment by chetanr2512** [comment]
yeah got i I am trying with a different account and a number

**Comment by chetanr2512** [comment]
can you also tell what was the use of tokens here
![Screenshot 2023-04-14 234655](https://user-images.githubusercontent.com/130687370/232125130-323bdd12-3bc7-4ab7-8d29-f0de5e7e89d6.png)
is it like how many characters?

**Comment by chetanr2512** [comment]
ok got It bro its the number of character . Also one more request how can we make the prompt like that so it only shows in 1000 
words 
like currently it is showing like this after the limit has exceeded
![Screenshot 2023-04-14 235024](https://user-images.githubusercontent.com/130687370/232125801-94193c75-2d74-41d1-8afe-7ca9237c1ac0.png)
you can see the star right

**Comment by chetanr2512** [comment]
got It bro I changed the let prompt and added "under 1000 words"
 Really appreciate you and your work man and really thank you for your time
also 1 more thing
![Screenshot 2023-04-14 235419](https://user-images.githubusercontent.com/130687370/232126546-590a8e55-e0df-41ee-8560-65d35f06e32b.png)
 here it is select multiple options but you can only select one 
also in your deployment the selection it just gets blue and the dropdown menu does not close I made code changes so the dropdown menu dosent always shows but don't know how to make multiple selection

**Comment by zinedkaloc** [comment]
This is related to OpenAI tokens. You can find more information on how to use tokens in the documentation, located here: [OpenAI API - Tokenizer](https://platform.openai.com/tokenizer)

One possible answer to fix it to 1000 token or less you could add a 'shorten response' at the end of the prompt.

To help improve performance, I recommend reducing the number of tokens used by the prompt and optimizing it to consume minimal tokens. This should help decrease the load and improve the speed of the process.

**Comment by chetanr2512** [comment]
yeah thanks man I did that already still really appreciate your reply

**Comment by AbhinavChebrolu** [comment]
I am having error 400, which gives You exceeded your current quota; please check your plan and billing details.
Please help me out where we need to sort it out. I made new accounts in both ChatGPT and Altogic even though the same error was raised. Even tokens were reduced to 100 and 16.
please explain me how to proceed with this step @zinedkaloc @chetanr2512
